Compilation error on Fedora 40: incompatible pointer types

See https://simc.arpae.it/moncic-ci/Magics-rpm/202407181635/master/fedora40/build.log:

```
2024-07-18 19:00:46 INFO system.fedora40 stderr: /root/rpmbuild/BUILD/Magics-4.14.2-Source/src/drivers/minizip/zip.c:883:28: error: assignment to ‘const long unsigned int *’ from incompatible pointer type ‘const uint32_t *’ {aka ‘const unsigned int *’} [-Wincompatible-pointer-types]
2024-07-18 19:00:46 INFO system.fedora40 stderr:   883 |         zi->ci.pcrc_32_tab = get_crc_table();
2024-07-18 19:00:46 INFO system.fedora40 stderr:       |                            ^
```

--- a/Magics.spec
+++ b/Magics.spec
@@ -98,6 +98,7 @@ Header and library files for Magics - The library and tools to visualize meteoro
 # (see: https://jira.ecmwf.int/browse/SUP-3299)
 %cmake \
     -DCMAKE_PREFIX_PATH=%{_prefix} \
+    -DCMAKE_C_FLAGS="$CFLAGS -Wno-incompatible-pointer-types" \
     -DCMAKE_CXX_FLAGS="$CXXFLAGS -I/usr/include/libgeotiff" \
     -DCMAKE_INSTALL_PREFIX=%{_prefix} \
     -DCMAKE_INSTALL_MESSAGE=NEVER \

